apps/full-site-editing/full-site-editing-plugin/full-site-editing/templates/class-wp-template-inserter.php

```diff
@@ -157,6 +166,99 @@ public function insert_default_template_data() {
 		add_option( $this->fse_template_data_option, true );
 	}
 
+	/**
+	 * Determines whether default pages have already been created.
+	 *
+	 * @return bool True if default pages have already been created, false otherwise.
+	 */
+	public function is_pages_data_inserted() {
+		return get_option( $this->fse_page_data_option ) ? true : false;
+	}
+
+	/**
+	 * Inserts default About and Contact pages based on Starter Page Templates content.
+	 *
+	 * The insertion will not happen if this data has been already inserted or if pages
+	 * with 'About' and 'Contact' titles already exist.
+	 */
+	public function insert_default_pages() {
+		// Bail if this data has already been inserted.
+		if ( $this->is_pages_data_inserted() ) {
+			return;
+		}
+
+		$request_url = add_query_arg(
+			[
+				'_locale' => $this->get_iso_639_locale(),
+			],
+			'https://public-api.wordpress.com/wpcom/v2/verticals/m1/templates'
+		);
+
+		$response = wp_remote_get( $request_url );
```

Comment:
Could we re-use the SPT implementation of this?

Reply:
I looked into doing that but it would iirc it would involve initializing SPT class, which registers various actions and filters. In principle this should be isolated from it - I'd view that as a very unexpected side effect (essentially turning SPTs on), unless we refactor that out in a separate class that should be moved out of SPT or FSE folder tree since it's shared. I might be misremembering though, would you mind linking to the code you had in mind?